src/v/redpanda/application.cc

```diff
@@ -1290,56 +1228,81 @@ application::set_proxy_client_config(ss::sstring name, std::any val) {
-    start_kafka(app_signal);
+void application::wire_up_bootstrap_services() {
+    // Wire up local storage.
+    ss::smp::invoke_on_all([] {
+        return storage::internal::chunks().start();
+    }).get();
+    syschecks::systemd_message("Constructing storage services").get();
+    construct_single_service_sharded(storage_node).get();
+    construct_service(
+      storage,
+      []() { return kvstore_config_from_global_config(); },
+      [this]() {
+          auto log_cfg = manager_config_from_global_config(_scheduling_groups);
+          log_cfg.reclaim_opts.background_reclaimer_sg
+            = _scheduling_groups.cache_background_reclaim_sg();
+          return log_cfg;
+      })
+      .get();
 
-    _admin.invoke_on_all([](admin_server& admin) { admin.set_ready(); }).get();
+    // Wire up the internal RPC server.
+    ss::sharded<net::server_configuration> rpc_cfg;
+    rpc_cfg.start(ss::sstring("internal_rpc")).get();
+    rpc_cfg
+      .invoke_on_all([this](net::server_configuration& c) {
+          return ss::async([this, &c] {
+              auto rpc_server_addr
+                = net::resolve_dns(config::node().rpc_server()).get0();
+              c.load_balancing_algo
+                = ss::server_socket::load_balancing_algorithm::port;
+              c.max_service_memory_per_core = memory_groups::rpc_total_memory();
+              c.disable_metrics = net::metrics_disabled(
+                config::shard_local_cfg().disable_metrics());
+              c.disable_public_metrics = net::public_metrics_disabled(
+                config::shard_local_cfg().disable_public_metrics());
+              c.listen_backlog
+                = config::shard_local_cfg().rpc_server_listen_backlog;
+              c.tcp_recv_buf
+                = config::shard_local_cfg().rpc_server_tcp_recv_buf;
+              c.tcp_send_buf
+                = config::shard_local_cfg().rpc_server_tcp_send_buf;
+              auto rpc_builder = config::node()
+                                   .rpc_server_tls()
+                                   .get_credentials_builder()
+                                   .get0();
+              auto credentials
+                = rpc_builder
+                    ? rpc_builder
+                        ->build_reloadable_server_credentials(
+                          [this](
+                            const std::unordered_set<ss::sstring>& updated,
+                            const std::exception_ptr& eptr) {
+                              cluster::log_certificate_reload_event(
+                                _log, "Internal RPC TLS", updated, eptr);
+                          })
+                        .get0()
+                    : nullptr;
+              c.addrs.emplace_back(rpc_server_addr, credentials);
+          });
+      })
+      .get();
 
-    vlog(_log.info, "Successfully started Redpanda!");
-    syschecks::systemd_notify_ready().get();
+    // Use port based load_balancing_algorithm to make connection shard
```

Comment:
nit: i know that this was moved directly from other method, but can we move this comment to be above line 1257 so that it matches the actual line where we set balancing algo ?

Reply:
Nice catch! Done